src/contract_client/client.ts

```diff
@@ -18,31 +18,59 @@ export class ContractClient {
     let methods = this.spec.funcs();
     for (let method of methods) {
       let name = method.name().toString();
-      // @ts-ignore
-      this[name] = async (
-        args: Record<string, any>,
-        options: MethodOptions
-      ) => {
-        return await AssembledTransaction.build({
-          method: name,
-          args: spec.funcArgsToScVals(name, args),
-          ...options,
-          ...this.options,
-          errorTypes: spec
-            .errorCases()
-            .reduce(
-              (acc, curr) => ({
-                ...acc,
-                [curr.value()]: { message: curr.doc().toString() },
-              }),
-              {} as Pick<ContractClientOptions, "errorTypes">
-            ),
-          parseResultXdr: (result: xdr.ScVal) => spec.funcResToNative(name, result),
-        });
-      };
+      let fn = spec.getFunc(name);
+
+      if (fn.inputs().length === 0) {
+        // @ts-ignore error TS7053: Element implicitly has an 'any' type
+        this[name] = async (methodOptions?: MethodOptions) => {
+          return await ContractClient.assembleTransaction(
+            name,
+            spec,
+            options,
+            undefined,
+            methodOptions,
+          );
+        };
+      } else {
+        // @ts-ignore error TS7053: Element implicitly has an 'any' type
+        this[name] = async (args: Record<string, any>, methodOptions?: MethodOptions) => {
+          return await ContractClient.assembleTransaction(
+            name,
+            spec,
+            options,
+            args,
+            methodOptions,
+          );
+        };
+      }
```

Comment:
the action is identical, right? as a suggestion, this is mildly cleaner:

```suggestion
        const action = async (args: Record<string, any>, methodOptions?: MethodOptions) => {
          return await ContractClient.assembleTransaction(
            name,
            spec,
            options,
            args,
            methodOptions,
          );

      if (fn.inputs().length === 0) {
        // @ts-ignore error TS7053: Element implicitly has an 'any' type
        this[name] = async (methodOptions?: MethodOptions) => {
          return await action(undefined, methodOptions);
        }
      } else {
        this[name] = action;
      }
```

Reply:
Oh, I like that, but that gave me an idea for an even slicker implementation (I think). I also switched to an editor ([Zed](https://registerspill.thorstenball.com/p/from-vim-to-zed)) that does a better job of showing me eslint errors, and noticed eslint was upset about the `Pick` in this file. So I switched the parser to `@typescript-eslint/parser`. lmk what you think!

Reply:
noice!